src/dbwrapper.cpp

```diff
@@ -71,14 +71,41 @@ class CBitcoinLevelDBLogger : public leveldb::Logger {
     }
 };
 
+constexpr int PermissibleMaxOpenFiles() {
+#ifdef _POSIX_C_SOURCE
```

Comment:
I don't think this does what you expect it to. As I understand it, confirmed by the [gcc page](https://www.gnu.org/software/libc/manual/html_node/Feature-Test-Macros.html) about it, `_POSIX_C_SOURCE` is supposed to be defined explicitly at the top of the compilation unit to request certain POSIX C library features. It is not defined by the compiler. So it can't be used to detect whether the platform compiled to is POSIX.

In other place, we've just used `#if[n]def WIN32` as WIN32 is the only non-POSIX platform supported in this project.

Reply:
You're absolutely right, I will fix this.